text/0000-remove-application-wrapper.md

```diff
@@ -0,0 +1,99 @@
+`#ember-testing` container for no good reason.
+
+This RFC proposes to add a global flag to remove those wrapper elements,
+effectively making the `application.hbs` template have "Outer HTML" semantics, which aligns
+well with [the changes proposed recently](#278) for template-only components and the way glimmer
+apps work.
+
+The same flag will also remove the unnecessary extra wrapper inside the testing
+container.
+
+# Detailed design
+
+## API Surface
+
+The proposed approach is identical to the one proposed in #278, that I reproduce
+pristine below in italics:
+
+*We should not expose the flag directly as a public API. Instead, we should
+abstract the flag with a "privileged addon" whose only purpose is to enable
+the flag. Applications will enable the flag by installing this addon. This
+will allow for more flexibility in changing the flag's implementation (the
+location, naming, value, or even the existence of it) in the future. From the
+user's perspective, it is the addon that provides this functionality. The
+flag is simply an internal implementation detail.*
+
+*We have done this before in other cases (such as the legacy view addon during
+the 2.0 transition period), and it has generally worked well.*
+
+*When landing this feature, it will be entirely opt-in for existing apps, but
+the Ember CLI application blueprint should be updated to include the addon by
+default. At a later time, we should provide another addon that _disables_ the
+flag explicitly (installing both addons would be an install-time error). At
+that time, we will issue a deprecation warning if the flag is *not set*, with
+a message that directs the user to install one of the two addons.*
+
+## Migration Path
+
+Given that this change only affects one single point in your application,
+I do not believe we need any specific strategy. If the users want to bring
+back the wrapper because it breaks their styles or some other reason,
+they can just add it manually on the `application.hbs` template, with
+any class or id they want.
+
+# How We Teach This
+
+This addon will be an opt in, but at some point it will become part of
+the default blueprint. This change, rather than introducing a new concept, *removes*
+an old one. Users won't have to google what is the way to remove or customize
+the implicit application wrapper of the app (to sadly discover that is not
+even possible), but instead they will just add a wrapper only if they want,
+and in the same way they would add a wrapper in any other point of their application,
+with regular handlebars.
+
+# Drawbacks
+
+There is a possibility that removing the wrapper can break styles for some apps,
+but since adding the wrapper back is just of editing the `application.hbs` template,
+that is probably a despicable drawback.
+
+There is also a non-zero chance that some testing addon is relying on the `#ember-testing > .ember-view`
+html hierarchy for some reason, and those addons would have to be updated.
```

Comment:
FWIW qunit-dom supports both cases already via fallback if no ember-view child is found